src/main/java/dev/slimevr/filtering/QuaternionMovingAverage.java

```diff
@@ -0,0 +1,84 @@
+package dev.slimevr.filtering;
+
+import com.jme3.math.FastMath;
+import com.jme3.math.Quaternion;
+
+
+public class QuaternionMovingAverage {
+
+	private final float smoothFactor;
+	private final float predictFactor;
+	private final CircularArrayList<Quaternion> rotBuffer;
+	private final Quaternion quatBuf = new Quaternion();
+	private final Quaternion targetQuat = new Quaternion();
+	private final Quaternion lastQuaternion;
+	private final Quaternion filteredQuaternion;
+	private static final float SMOOTH_QUADRATIC_DIVIDER = 11f;
+	private static final float SMOOTH_QUADRATIC_MIN = 0.022f;
+	private static final float PREDICT_QUADRATIC_DIVIDER = 3.1f;
+	private static final float PREDICT_QUADRATIC_MIN = 0.11f;
+	private static final int PREDICT_BUFFER = 6;
+
+
+	public QuaternionMovingAverage(
+		TrackerFilters type,
+		float amount,
+		Quaternion initialRotation
+	) {
+		amount = Math.max(amount, 0);
```

Comment:
This makes it clear the minimum value is 0, what's the maximum value, 1? I would assume 1 since it's being used in the interpolation, but maybe a little comment explicitly stating that could be helpful.